### PR TITLE
Add Before and After for cypress-cucumber-preprocessor

### PR DESCRIPTION
**Pull request by FalconerTC**
cypress-cucumber-preprocessor added support for Cypress hooks Before and After in 1.14

### CODE REVIEW

File: types/cypress-cucumber-preprocessor/steps/index.d.ts [hunk 1]
```diff
@@ -12,3 +12,5 @@ export function When(expression: RegExp | string, implementation: (...args: any[
 export function Then(expression: RegExp | string, implementation: (...args: any[]) => void): void;
 export function And(expression: RegExp | string, implementation: (...args: any[]) => void): void;
 export function But(expression: RegExp | string, implementation: (...args: any[]) => void): void;
+export function Before(args_0: object | ((...args: any[]) => void), args_1?: (...args: any[]) => void): void;
```
Comment (aleccool213) [inline types/cypress-cucumber-preprocessor/steps/index.d.ts]: Can you link to where this code is defined in the original repo with a reply to this comment? I have an idea why `args_0` is defined this way but don't understand `args_1`.

Comment (FalconerTC) [reply types/cypress-cucumber-preprocessor/steps/index.d.ts]: It's use is documented [here](https://github.com/cucumber/cucumber-js/blob/master/docs/support_files/hooks.md). I'm new to Typescript so let me know if this is not the correct bindings interpretation.

Comment (FalconerTC) [reply types/cypress-cucumber-preprocessor/steps/index.d.ts]: The function that parses this in cypress-cucumber-preprocessor is [here](https://github.com/TheBrainFamily/cypress-cucumber-preprocessor/blob/master/lib/resolveStepDefinition.js#L153)

Comment (aleccool213) [reply types/cypress-cucumber-preprocessor/steps/index.d.ts]: Thanks. This looks correct!

I dislike the Cucumber api for `Before/After` due to is optional first argument of `options`. This made it difficult to know if your code was correct or not, not your fault. Let's name the arguments to something more specific.

```typescript
export function Before(optionsOrImplementation: object | ((...args: any[]) => void), implementation?: (...args: any[]) => void): void;
```

**Review by aleccool213 — CHANGES_REQUESTED** [review]

**Review by aleccool213 — APPROVED** [review]

**Pull request merged**